bindings/python/tests/test_datafusion_table_provider.py

```diff
@@ -0,0 +1,160 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+from datetime import date, datetime
+import uuid
+import pytest
+from pyiceberg_core.datafusion import IcebergDataFusionTable
+from datafusion import SessionContext
+from pyiceberg.catalog import Catalog, load_catalog
+import pyarrow as pa
+from pathlib import Path
+import datafusion
+
+assert (
+    datafusion.__version__ >= "45"
```

Comment:
Do we need to make sure that rust and python are using the same version of datafusion? 

cc @alamb for ideas.